doc/feature/genericEventLoading.md

```diff
@@ -0,0 +1,119 @@
+a set of events in 3D space. For example, in a neural circuit with synaptic
+connectivity, we are interested in visualizing the position of the post-synaptic
+cells; or for validation purposes, we want to check that the LFP computation is
+done correctly for a predefined set of events.
+
+Since the variety of use cases is so wide, each of them with its own
+particularities, we need to find a generic way of loading any predefined set of
+events.
+
+The solution chosen for this feature is the loading of events from file. This
+offers users flexibility to generate all types of events in a simple and generic
+way, so the tool doesn't need to be adapted to fit the particular needs of any
+situation.
+
+
+## Requirements
+
+* Users should be able to specify their own input events, as easily as possible,
+from an external source (i.e. file).
+* The format of the input file should be clearly defined and documented.
+* All the library common capabilities should still be available for their use 
+with this feature. For example, generating 3D volumes with different
+resolutions and using any available functor.
+
+
+## API
+
+A TestLoader is already present in Fivox, which generates a set of 7 events in a
+fixed position, and is used for basic use cases such as the validation of the
+LFP computation in a test case. The input URI in this case is as follows:
+
+    "fivoxtest://?resolution=1&functor=lfp"
+
+What we need for the loading of any number of generic events is very similar to
+this, but we need a way to specify the source of our events, instead of
+generating them at runtime. We can use the BlueConfig path location as the event
+file path. A change in the name might also be convenient, as it is not anymore
+just for testing purposes. The loader and schema names are open for discussion.
+
+    "fivox:///absolute/path/to/events/file?resolution=1&functor=lfp"
+
+The path to the input file is optional, behaving like the original TestLoader
+and creating the initial 7 events if it is omitted.
+
+For simplicity, and for now, only the GenericLoader can read events from file.
+But any EventSource can write this type of files, dumping their contents (all
+the events and their corrresponding values that it contains at the moment).
+For that, a new method will be added:
+
+    bool EventSource::write( const std::string& filename, bool binary = true );
+
+The corresponding option should also be added in the _voxelize_ application,
+so users have access to it.
+
+
+## File format
+
+For the input file containing the events, two different formats should be
+supported, binary for performance, and ASCII for readability, but keeping the
+same layout.
+
+#### ASCII
+
+Header containing the version numbers and the format used, followed by all the
+events (one event per line).
+
+Example:
+
+```cpp
+# Fivox events (3D position, radius and value), in the following format:
+#     posX posY posZ radius value
+# File version: 1
+# Fivox version: 0.6.0
```

Comment:
I already used this header type in another tool (compute-vsd), for several output files. I'd keep it like this for consistency.